### PR TITLE
Add module-info to jca and jca test modules.

### PR DESCRIPTION
**Pull request by michaelqi793**
Also #21426

### CODE REVIEW

File: sdk/keyvault/azure-security-test-keyvault-jca/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * This is for testing the modularization of module azure.security.keyvault.jca
+ */
+module azure.security.test.keyvault.jca {
+    requires azure.security.keyvault.jca;
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-test-keyvault-jca/src/main/java/module-info.java:8]: Did you test this:
if there is something wrong in `sdk/keyvault/azure-security-keyvault-jca/src/main/module-info.java`?
will pipeline failure?

**Comment by chenrujun** [comment]
Hi, @michaelqi793 , please merge latest master branch to your feature branch.

**Comment by chenrujun** [comment]
So pipeline passed after [minor change](https://github.com/Azure/azure-sdk-for-java/pull/21965/commits/6771ffb1d831c44c67ef2db8d176843c353e8c75).

**Comment by michaelqi793** [comment]
Replaced with #22041

**Pull request closed without merging**